src/EventListener/Doctrine/SynonymListener.php

```diff
@@ -30,12 +34,31 @@ public function postRemove(LifecycleEventArgs $eventArgs): void
         $this->handle($eventArgs);
     }
 
+    /**
+     * This method can be called multiple times in the same request, therefore we set a flag to only dispatch the command once
+     */
     public function handle(LifecycleEventArgs $eventArgs): void
     {
         if (!$eventArgs->getObject() instanceof SynonymInterface) {
             return;
         }
 
+        $this->update = true;
+    }
+
+    public function dispatch(): void
+    {
+        if (!$this->update) {
+            return;
+        }
+
         $this->commandBus->dispatch(new UpdateSynonyms());
     }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::RESPONSE => ['dispatch', 10],
```

Comment:
@Zales0123 @jakubtobiasz: You you also listen to the `RESPONSE` or would you go for another event?

Reply:
Why don't you listen to resource events?

Reply:
I am not sure I know what resource events to listen to? I mean the indexes are configurable, so I don't know the entities/resources used to build an index.

Reply:
Are they also fired when using the API?